aspnetcore/mvc/views/partial.md

```diff
@@ -1,173 +1,280 @@
+<xref:Microsoft.AspNetCore.Mvc.Rendering.HtmlHelperPartialExtensions.Partial*> and <xref:Microsoft.AspNetCore.Mvc.Rendering.HtmlHelperPartialExtensions.RenderPartial*> are the synchronous equivalents of `PartialAsync` and `RenderPartialAsync`, respectively. The synchronous equivalents aren't recommended because there are scenarios in which they deadlock. The synchronous methods are targeted for removal in a future release.
 
 > [!IMPORTANT]
-> If your views need to execute code, use a [view component](xref:mvc/views/view-components) instead of a partial view.
+> If you need to execute code, use a [view component](xref:mvc/views/view-components) instead of a partial view.
 
 ::: moniker range=">= aspnetcore-2.1"
 
-In ASP.NET Core 2.1 or later, calling `Partial` or `RenderPartial` results in an analyzer warning. For example, usage of `Partial` yields the following warning message:
+Calling `Partial` or `RenderPartial` results in a Visual Studio analyzer warning. For example, the presence of `Partial` yields the following warning message:
 
-> Use of IHtmlHelper.Partial may result in application deadlocks. Consider using `<partial>` Tag Helper or `IHtmlHelper.PartialAsync`.
+> Use of IHtmlHelper.Partial may result in application deadlocks. Consider using &lt;partial&gt; Tag Helper or IHtmlHelper.PartialAsync.
 
-Replace calls to `@Html.Partial` with `@await Html.PartialAsync` or the Partial Tag Helper. For more information on Partial Tag Helper migration, see [Migrate from an HTML Helper](xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper#migrate-from-an-html-helper).
+Replace calls to `@Html.Partial` with `@await Html.PartialAsync` or the [Partial Tag Helper](xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper). For more information on Partial Tag Helper migration, see [Migrate from an HTML Helper](xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper#migrate-from-an-html-helper).
 
 ::: moniker-end
 
 ## Partial view discovery
 
-When referencing a partial view, you can refer to its location in several ways. For example:
+When a partial view is referenced by name without a file extension, the following locations are searched in the stated order:
 
 ::: moniker range=">= aspnetcore-2.1"
 
-```cshtml
-// Uses a view in current folder with this name.
-// If none is found, searches the Shared folder.
-<partial name="_ViewName" />
-
-// A view with this name must be in the same folder
-<partial name="_ViewName.cshtml" />
-
-// Locate the view based on the app root.
-// Paths that start with "/" or "~/" refer to the app root.
-<partial name="~/Views/Folder/_ViewName.cshtml" />
-<partial name="/Views/Folder/_ViewName.cshtml" />
+**Razor Pages**
 
-// Locate the view using a relative path
-<partial name="../Account/_LoginPartial.cshtml" />
-```
+1. Currently executing page's folder
```

Comment:
Sorry, should have caught this - for a Razor Page, this includes the directory graph. `/Pages` would happen to be in it by accident. For instance, doing `<partial name="_Deals" />` from `/Pages/Products/Electronics/TV.cshtml` would look for the partial in:

1. `/_Pages/Produces/Electronics/_Deals.cshtml`
1. `/_Pages/Produces/_Deals.cshtml`
1. `/_Pages/_Deals.cshtml`
1. `/Pages/Shared/_Deals.cshtml`
1. `/Views/Shared/_Deals.cshtml`

https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Razor/RazorViewEngineOptions.cs#L125-L129 has the docs for the lookup semantics

Reply:
Is the same true for MVC views? ... does it also go up the directory graph?

Reply:
Nope. Just a Razor Pages thing.

Reply:
Ok ... I left it that way on the last commit, which has the changes for Pages and the Areas coverage.